//
// Copyright 2020 The Feast Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//


syntax = "proto3";
package feast.core;

option go_package = "github.com/feast-dev/feast/go/protos/feast/core";
option java_outer_classname = "DataFormatProto";
option java_package = "feast.proto.core";

// Defines the file format encoding the features/entity data in files
message FileFormat {
  // Defines options for the Parquet data format
  message ParquetFormat {}
  
  oneof format {
    ParquetFormat parquet_format = 1;
  }
}

// Defines the data format encoding features/entity data in data streams
message StreamFormat {
  // Defines options for the protobuf data format
  message ProtoFormat {
    // Classpath to the generated Java Protobuf class that can be used to decode
    // Feature data from the obtained stream message
    string class_path = 1;
  }
  
  // Defines options for the avro data format
  message AvroFormat {
    // Optional if used in a File DataSource as schema is embedded in avro file.
    // Specifies the schema of the Avro message as JSON string.
    string schema_json = 1;
  }

  // Specifies the data format and format specific options
  oneof format {
    AvroFormat avro_format = 1;
    ProtoFormat proto_format = 2;
  }
}
